net/wgpeerselector/files/usr/bin/wgpeerselector

```diff
@@ -0,0 +1,505 @@
+	return o
+end
+
+function WGPeer:get_next_remote()
+	local ipv4_endpoint = string.match(self.endpoint, "^(%d+%.%d+%.%d+%.%d+:%d+)$")
+	if ipv4_endpoint then
+		return ipv4_endpoint
+	end
+
+	local ipv6_endpoint = string.match(self.endpoint, "^(%[[%x:]+%]:%d+)$")
+	if ipv6_endpoint then
+		return ipv6_endpoint
+	end
+
+	-- We cycle round robin through the dns responses. Therefore we keep the
+	-- results of the dns responses in self.next_remotes and pop one randomly
+	-- each time. If self.next_remotes is empty, do dns resolution again.
+	if #self.next_remotes < 1 then
+		-- do dns resolution
+		local host, port = string.match(self.endpoint, "^([%w-.]+):(%d+)$")
+		if not host or not port then
+			log(syslog.LOG_ERR, "Endpoint '"..self.endpoint.."' is not a valid endpoint.")
+			return nil
+		end
+
+		local res, _, errcode = sys_sock.getaddrinfo(host, nil, { socktype = sys_sock.SOCK_DGRAM })
+
+		if errcode then
+			log(syslog.LOG_WARNING, "DNS resolution of " .. self.endpoint .. " failed: error code " .. tostring(errcode))
+			return nil
+		end
+
+		for i = 1, #res do
+			if string.match(res[i].addr, ":") then -- IPv6
+				table.insert(self.next_remotes, '['..res[i].addr..']:'..port)
+			else -- IPv4
+				table.insert(self.next_remotes, res[i].addr..':'..port)
+			end
+		end
+	end
+
+	return table.remove(self.next_remotes, math.random(#self.next_remotes))
+end
+
+function WGPeer:install_to_kernel()
+	local remote = self:get_next_remote()
+	if not remote then
+		return false
+	end
+
+	local command = "wg set '" .. self.iface .. "' peer '" .. self.public_key .. "' " ..
+	                "allowed-ips '" .. table.concat(self.allowed_ips, ",") .. "' " ..
+	                "endpoint '" .. remote .. "' " ..
+	                "persistent-keepalive '" .. tostring(self.persistent_keepalive) .. "'"
+	log(syslog.LOG_INFO, "Adding peer " .. self.name .. " to " .. self.iface .. ".")
+	if os.execute(command) ~= 0 then
+		log(syslog.LOG_ERR, "Couldn't add peer " .. self.name .. " to " .. self.iface .. ".")
+		return false
+	end
+
+	-- install routes
+	for _, allowed_ip in pairs(self.allowed_ips) do
+		if os.execute("ip route add '" .. allowed_ip .. "' dev '" .. self.iface .. "'") ~= 0 then
+			log(syslog.LOG_WARNING, "Couldn't add route " .. allowed_ip .. " to " .. self.iface .. ".")
+		end
+	end
+
+	return true
+end
+
+function WGPeer:uninstall_from_kernel()
+	local command = "wg set '" .. self.iface  .. "' peer '" .. self.public_key .. "' remove"
+	log(syslog.LOG_INFO, "Removing peer " .. self.name .. " from " .. self.iface ..".")
+	if os.execute(command) ~= 0 then
+		log(syslog.LOG_WARNING, "Couldn't remove peer " .. self.name .. " from " .. self.iface .. ".")
+	end
+	for _, allowed_ip in pairs(self.allowed_ips) do
+		if os.execute("ip route del '" .. allowed_ip .. "' dev '" .. self.iface .. "'") ~= 0 then
+			log(syslog.LOG_WARNING, "Couldn't del route " .. allowed_ip .. " from " .. self.iface .. ".")
+		end
+	end
+	self.established_at = 0
+end
```

Comment:
`WGPeer:install_to_kernel()` returns boolean error codes, so should this function.